plaso/storage/redis/reader.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+"""Redis reader."""
+from __future__ import unicode_literals
+
+from plaso.lib import definitions
+from plaso.storage.redis import redis_store
+from plaso.storage import interface
+
+class RedisStorageReader(interface.StorageReader):
+  """Redis storage file reader."""
+
+  def __init__(self, task):
+    super(RedisStorageReader, self).__init__()
+    self._store = redis_store.RedisStore(
+        storage_type=definitions.STORAGE_TYPE_TASK,
+        session_identifier=task.session_identifier,
+        task_identifier=task.identifier)
+
+  def IsFinalized(self):
+    """Checks if the store has been finalized.
+
+    Returns:
+      bool: True if the store has been finalized.
+    """
+    return self._store.IsFinalized()
+
+  def Open(self):
+    """Opens the storage reader."""
+    self._store.Open()
+
+  def Close(self):
+    """Closes the storage reader."""
+    self._store.Close()
+
+  def GetAnalysisReports(self):
+    """Retrieves the analysis reports.
+
+    Yields:
+      AnalysisReport: analysis report.
+    """
+    for report in self._store.GetAnalysisReports():
```

Comment:
Based on `plaso/storage/interface.py` `self._store.GetAnalysisReports()` returns a generator, why not just return this?